app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchSuggestionsFragment.kt

```diff
@@ -182,110 +158,61 @@ class SearchSuggestionsFragment : Fragment(), CoroutineScope {
         fun create() = SearchSuggestionsFragment()
     }
+}
 
-    inner class SuggestionsAdapter(
-        private val clickListener: (String) -> Unit
-    ) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
-        inner class DiffCallback(
-            private val oldSuggestions: List<SpannableStringBuilder>,
-            private val newSuggestions: List<SpannableStringBuilder>
-        ) : DiffUtil.Callback() {
-            override fun getOldListSize(): Int = oldSuggestions.size
-            override fun getNewListSize(): Int = newSuggestions.size
-            override fun areItemsTheSame(p0: Int, p1: Int): Boolean = true
-            override fun areContentsTheSame(p0: Int, p1: Int): Boolean =
-                    oldSuggestions[p0] == newSuggestions[p1]
-            override fun getChangePayload(oldItemPosition: Int, newItemPosition: Int): Any? =
-                    newSuggestions[newItemPosition]
-        }
-
-        private var suggestions: List<SpannableStringBuilder> = listOf()
-        private var pendingJob: Job? = null
-
-        suspend fun refresh(suggestions: List<SpannableStringBuilder>) = coroutineScope {
-            pendingJob?.cancel()
-
-            pendingJob = launch(IO) {
-                val result = DiffUtil.calculateDiff(DiffCallback(this@SuggestionsAdapter.suggestions, suggestions))
-
-                launch(Main) {
-                    result.dispatchUpdatesTo(this@SuggestionsAdapter)
-                    this@SuggestionsAdapter.suggestions = suggestions
-                }
-            }
-        }
-
-        override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int, payloads: MutableList<Any>) {
-            if (payloads.isEmpty()) {
-                super.onBindViewHolder(holder, position, payloads)
-            } else {
-                val payload = payloads[0] as? SpannableStringBuilder ?: return
-                val view = holder as? SuggestionViewHolder ?: return
-                view.bind(payload)
-            }
-        }
-
-        override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
-            if (holder is SuggestionViewHolder) {
-                holder.bind(suggestions[position])
-            }
-        }
-
-        override fun getItemCount(): Int = suggestions.count()
-
-        override fun getItemViewType(position: Int): Int {
-            return SuggestionViewHolder.LAYOUT_ID
-        }
-
-        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
-            return SuggestionViewHolder(
-                    LayoutInflater.from(parent.context).inflate(viewType, parent, false),
-                    clickListener)
-        }
+@Composable
+private fun SearchOverlay(
+    viewModel: SearchSuggestionsViewModel
+) {
+    val state = viewModel.state.observeAsState()
+
+    when (state.value) {
+        is State.Disabled -> { /* Not implemented in Compose, still handled by the View implementation */ }
```

Comment:
Is the intent here to eventually transition the other states over to this Composeable?

Reply:
Yeah, exactly. I didn't want to rewrite all those prompts for now.